src/query/on_conflict.rs

```diff
@@ -64,8 +64,84 @@ impl OnConflict {
         }
     }
 
+    /// Set ON CONFLICT do nothing.
+    /// Please use do_nothing_mysql() and provide pkcols if you are using MySql
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use sea_query::{tests_cfg::*, *};
+    ///
+    /// let query = Query::insert()
+    ///     .into_table(Glyph::Table)
+    ///     .columns([Glyph::Aspect, Glyph::Image])
+    ///     .values_panic(["abcd".into(), 3.1415.into()])
+    ///     .on_conflict(
+    ///         OnConflict::columns([Glyph::Id, Glyph::Aspect])
+    ///             .do_nothing()
+    ///             .to_owned(),
+    ///     )
+    ///     .to_owned();
+    ///
+    /// assert_eq!(
+    ///     query.to_string(PostgresQueryBuilder),
+    ///     [
+    ///         r#"INSERT INTO "glyph" ("aspect", "image")"#,
+    ///         r#"VALUES ('abcd', 3.1415)"#,
+    ///         r#"ON CONFLICT ("id", "aspect") DO NOTHING"#,
+    ///     ]
+    ///     .join(" ")
+    /// );
+    /// assert_eq!(
+    ///     query.to_string(SqliteQueryBuilder),
+    ///     [
+    ///         r#"INSERT INTO "glyph" ("aspect", "image")"#,
+    ///         r#"VALUES ('abcd', 3.1415)"#,
+    ///         r#"ON CONFLICT ("id", "aspect") DO NOTHING"#,
+    ///     ]
+    ///     .join(" ")
+    /// );
+    /// ```
     pub fn do_nothing(&mut self) -> &mut Self {
-        self.action = Some(OnConflictAction::DoNothing);
+        self.action = Some(OnConflictAction::DoNothing(vec![]));
+        self
+    }
+
+    /// Set ON CONFLICT do nothing specifically for MySql.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use sea_query::{tests_cfg::*, *};
+    ///
+    /// let query = Query::insert()
+    ///     .into_table(Glyph::Table)
+    ///     .columns([Glyph::Aspect, Glyph::Image])
+    ///     .values_panic(["abcd".into(), 3.1415.into()])
+    ///     .on_conflict(
+    ///         OnConflict::columns([Glyph::Id, Glyph::Aspect])
+    ///             .do_nothing_mysql(vec![Glyph::Id])
+    ///             .to_owned(),
+    ///     )
+    ///     .to_owned();
+    ///
+    /// assert_eq!(
+    ///     query.to_string(MysqlQueryBuilder),
+    ///     [
+    ///         r#"INSERT INTO `glyph` (`aspect`, `image`)"#,
+    ///         r#"VALUES ('abcd', 3.1415)"#,
+    ///         r#"ON DUPLICATE KEY UPDATE `id` = VALUES(`id`)"#,
+    ///     ]
+    ///     .join(" ")
+    /// );
+    pub fn do_nothing_mysql<C, I>(&mut self, pk_cols: I) -> &mut Self
```

Comment:
@darkmmon  I'm not sure, that naming is good... 

@tyt2y3 what do you think?

Reply:
Yes, may be `do_nothing_on` and add `MySQL only` in the doc? https://github.com/SeaQL/sea-query/blob/336bf954b94c618daef0b5c15a0dbf65da12ea25/src/table/column.rs#L384

Reply:
I agree with it